### ISSUE TITLE
Training loss follows a strange dynamics

### ISSUE THREAD

**Issue by vkedlian** [YosefLab/scvi-tools#1093]
Hi! Thanks for the wonderful set of tools!I have a question about the weird shape of the training loss while training scvi model for data integration . It seems to be decreasing at first, but then increasing again. See the screenshots attached (the dynamics of change is shown after first and then starting from 20 epochs to emphasize the following increase in the loss). The severity of the increase later seems to be dependent on which and how many batch covariates I supply for integration. What would you advise in this case? Should I train it for longer/shorter or pick the set of covariates that minimize this bump later? Thank you very much in advance!

```python
[CODE HERE]
```

```pytb
[ERROR HERE]
```

#### Versions:

> VERSION

![Integration_training1](https://user-images.githubusercontent.com/22593315/124785225-970b5200-df3e-11eb-9d55-8d4d94661f30.png)
![Integration_training2](https://user-images.githubusercontent.com/22593315/124785238-996dac00-df3e-11eb-8cba-e45b71bf1346.png)
![Integration_training3](https://user-images.githubusercontent.com/22593315/124785246-9c689c80-df3e-11eb-8f4a-f26164a0bcd1.png)

**Comment by adamgayoso** [comment]
So I don't think there's much to worry about here. The reason being that the loss is equal to:

`reconstruction_loss + kl_weight * kl_loss`

The `kl_weight` by default goes from 0 to 1 linearly over 400 epochs. So it makes sense that the loss can increase, since it's almost a moving target. I would inspect the ELBO instead.

**Comment by vkedlian** [comment]
Thank you very much @adamgayoso ! I am new to neural network and VAE field so it is very good to know. The ELBO is, indeed,  consistently decreasing.  I have a few follow up  questions on using scvi model, see below - but happy to move them to  discourse section if they don't belong here: 1) what are the parameters that you would advise to inspect how well the model fits? 2) Is there a way to extract and compare training and validation ELBO? Would that be useful to inspect? 3) This is more of a broad question : Would it be appropriate to decide on the exact set of covariates to use based on how well scvi model can fit the data (if it is generally possible to tell the difference in the level of fitting between models with different covariates)? Thank you very much again and my apologies if some of these are very naive questions. 
With best wishes, 
Veronika

**Comment by adamgayoso** [comment]
We would appreciate these on discourse. Thanks! There are no naive questions!

**Comment by adamgayoso** [comment]
Closing due to inactivity, but feel free to raise questions on discourse.
